app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav.php

```diff
@@ -399,7 +400,11 @@ public function modifyData(array $data)
             foreach ($attributes as $attribute) {
                 if (null !== ($attributeValue = $this->setupAttributeData($attribute))) {
                     if ($attribute->getFrontendInput() === 'price' && is_scalar($attributeValue)) {
-                        $attributeValue = $this->formatPrice($attributeValue);
+                        if ($this->locator->getProduct()->getTypeId() !== ProductType::TYPE_BUNDLE
+                            || $attribute->getAttributeCode() !== ProductAttributeInterface::CODE_SPECIAL_PRICE
```

Comment:
So, we have the condition like "either if product type is bundle or the attribute is special price" which means that the condition is true for all special prices regardless of the product type?